pipeline.py

```diff
@@ -72,26 +88,76 @@ class TextProcessingPipeline:
     """
 
     def __init__(self, corpus_manager: CorpusManager):
-        pass
+        self.corpus_manager = corpus_manager
 
     def run(self):
         """
         Runs pipeline process scenario
         """
-        pass
+        articles = self.corpus_manager.get_articles().values()
+
+        for article in articles:
+            raw_text = article.get_raw_text()
+            tokens = self._process(raw_text)
+
+            cleaned_tokens = []
+
+            for token in tokens:
+                cleaned_tokens.append(token.get_cleaned())
+
+            article.save_as(' '.join(cleaned_tokens), ArtifactType.cleaned)
 
     def _process(self, raw_text: str):
         """
         Processes each token and creates MorphToken class instance
         """
-        pass
+
+        mystem = Mystem()
+        raw_text_analysis = mystem.analyze(raw_text)
+
+        tokens = []
+
+        for single_word_analysis in raw_text_analysis:
+            if single_word_analysis['text'].isalpha():
+                token = MorphologicalToken(single_word_analysis['text'])
+                tokens.append(token)
+
+        return tokens
 
 
 def validate_dataset(path_to_validate):
     """
     Validates folder with assets
     """
-    pass
+    if isinstance(path_to_validate, str):
+        path_to_validate = Path(path_to_validate)
+
+    if not path_to_validate.exists():
+        raise FileNotFoundError
+
+    if not path_to_validate.is_dir():
+        raise NotADirectoryError
+
+    if len(list(path_to_validate.glob('*'))) == 0:
+        raise EmptyDirectoryError
+
+    counter_txt = 0
+    counter_meta = 0
+
+    for file in path_to_validate.glob('*'):
+        if str(file)[-7:] == 'raw.txt':
```

Comment:
here and in other places, consider using `endswith` [method](https://docs-python.ru/tutorial/operatsii-tekstovymi-strokami-str-python/metod-str-endswith/): it is designed exactly for this purpose and your code will be more readable